content/blog/2019/root-transition/index.md

```diff
@@ -0,0 +1,142 @@
+
+If you are not currently using the mTLS feature in Istio and will not use it in the future,
+you are not affected and should only upgrade to 1.1.8 or later versions.
+
+If you are not currently using the mTLS feature in Istio and may use it in the future,
+you are recommended to follow the procedure listed below to upgrade.
+
+If you are currently using the mTLS feature in Istio with self-signed certificates,
+please follow the procedure and check whether you will be affected.
+
+## Procedure
+
+### Step 0. Check when your root certificate will expire
+
+Download this [script](TODO) on a machine that has kubectl access to the cluster.
+
+{{< text bash>}}
+$ ./root-transition.sh check
+...
+===YOU HAVE 30 DAYS BEFORE THE ROOT CERT EXPIRED!=====
+{{< /text >}}
+
+Please run the following steps to upgrade immediately to avoid system outages, before your root certificate expires.
+
+### Step 1. Do the root certificate transition
+
+Run the following command to do the root certificate transition.
+
+{{< text bash>}}
+$ ./root_transition.sh transition
+create new ca cert, with trust domain as  cluster.abc
+Wed Jun  5 19:11:15 PDT 2019 delete old ca secret
+secret "istio-ca-secret" deleted
+Wed Jun  5 19:11:15 PDT 2019 create new ca secret
+secret/istio-ca-secret created
+pod "istio-citadel-8574b88bcd-j7v2d" deleted
+Wed Jun  5 19:11:18 PDT 2019 restarted Citadel, checking status
+NAME                             READY     STATUS    RESTARTS   AGE
+istio-citadel-8574b88bcd-l2g74   1/1       Running   0          3s
+New root certificate:
+Certificate:
+    Data:
+        ...
+        Validity
+            Not Before: Jun  6 03:24:43 2019 GMT
+            Not After : Jun  3 03:24:43 2029 GMT
+        Subject: O = " cluster.abc"
+        ...
+{{< /text >}}
+
+After the transition, the Envoy sidecars may be hot-restarted. This may have some impact on your traffic.
+Please refer to 
+[here](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/hot_restart#arch-overview-hot-restart) and
+[here](https://blog.envoyproxy.io/envoy-hot-restart-1d16b14555b5) for more details.
+
+### Step 2. Verify the new workload certificates are generated
+
+Run the following command to verify that all the new certificates are generated. If not, wait a minute and run it again.
+
+{{< text bash>}}
+$ ./root_transition.sh verify
+This script checks the current root CA certificate is propagated to all the Istio-managed workload secrets in the cluster.                    
+Root cert MD5 is 8fa8229ab89122edba73706e49a55e4c
+Checking namespace: default
+  Secret default.istio.default is updated.
+  Secret default.istio.sleep is updated.
+Checking namespace: istio-system
+  Secret istio-system.istio.default is updated.
+  ...
+------All Istio keys and certificates are updated in secret!
+{{< /text >}}
+
+### Step 3. Install Istio 1.1.8
+
+Note that this step is necessary, to make sure the control plane components and sidecar Envoys all reload the new certificates and keys.
```

Comment:
Delete the words Note that this step is necesarry.

Add warning callout for the remainde e.g. `To ensure the control plane commponents and sidecar Envoys all reload new certificates, the following step is mandatory`

Reply:
Given that 1.1.8 upgrade should come first, after updating the CA - will control plane components require a manual restart, or will they dynamically reload their certs?

Reply:
good question - I don't have an immediate answer.

Reply:
Some components are able to reload their certs while others (like the sidecar-injector etc.) can't.
The updating to 1.1.8 will make sure to redeploy the control plane, so the new certificates will be loaded.